### ISSUE TITLE
sparksql sql file that fails AL01

### ISSUE THREAD

**Issue by gvozdvmozgu** [sqlfluff/sqlfluff#5994]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff fix failed on AL01 on sql file.

### Expected Behaviour

sqlfluff is able to fix the errors in the file.

### Observed Behaviour

WARNING: Fixes for AL01 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.

### How to reproduce

```sql
select * from (values (1, 2), (3, 4)) as (a, b);
```

python src/sqlfluff fix --dialect sparksql pg.sql

### Dialect

sparksql

### Version

sqlfluff, version 3.0.7

### Configuration

default

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by peterbud** [comment]
With the latest `sqlfluff` it parses and you don't get the above mentioned warning, but you get a different one:
```bash
sqlfluff fix test.sql -d sparksql
==== finding fixable violations ====
WARNING    Loop limit on fixes reached [10].                                                                                                                                                                    
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [5 unfixable linting violations found]
```
